torch/nn/modules/instancenorm.py

```diff
@@ -0,0 +1,169 @@
+from .batchnorm import _BatchNorm
+from .. import functional as F
+
+
+class _InstanceNorm(_BatchNorm):
+    def __init__(self, num_features, eps=1e-5, momentum=0.1, affine=False):
+        super(_InstanceNorm, self).__init__(
+            num_features, eps, momentum, affine)
+
+    def forward(self, input):
+        self._check_input_dim(input)
+
+        b, c = input.size(0), input.size(1)
+
+        # Repeat stored stats and affine transform params
+        running_mean = self.running_mean.repeat(b)
+        running_var = self.running_var.repeat(b)
+
+        weight, bias = None, None
+        if self.affine:
+            weight = self.weight.repeat(b)
+            bias = self.bias.repeat(b)
+
+        # Apply instance norm
+        input_reshaped = input.view(1, b * c, *input.size()[2:])
```

Comment:
might be worth making `input` contiguous right before this, as view will only work on contiguous inputs:
```
input = input.contiguous()
```